src/System.Net.Security/src/System/Net/FixedSizeReader.cs

```diff
@@ -4,163 +4,75 @@
 using System.IO;
-using System.Threading.Tasks;
+using System.Threading;
 
 namespace System.Net
 {
-    //
-    // The class is a simple wrapper on top of a read stream. It will read the exact number of bytes requested.
-    // It will throw if EOF is reached before the expected number of bytes is returned.
-    //
-    internal class FixedSizeReader
+    /// <summary>
+    /// The class is a simple wrapper on top of a read stream. It will read the exact number of bytes requested.
+    /// It will throw if EOF is reached before the expected number of bytes is returned.
+    /// </summary>
+    internal static class FixedSizeReader
     {
-        private static readonly AsyncCallback s_readCallback = new AsyncCallback(ReadCallback);
-
-        private readonly Stream _transport;
-        private AsyncProtocolRequest _request;
-        private int _totalRead;
-
-        public FixedSizeReader(Stream transport)
+        /// <summary>
+        /// Returns 0 on legitimate EOF or if 0 bytes were requested, otherwise reads as directed or throws.
+        /// Returns count on success.
+        /// </summary>
+        public static int ReadPacket(Stream transport, byte[] buffer, int offset, int count)
         {
-            _transport = transport;
-        }
-
-        //
-        // Returns 0 on legitimate EOF or if 0 bytes were requested, otherwise reads as directed or throws.
-        // Returns count on success.
-        //
-        public int ReadPacket(byte[] buffer, int offset, int count)
-        {
-            int tempCount = count;
+            int remainingCount = count;
             do
             {
-                int bytes = _transport.Read(buffer, offset, tempCount);
-
+                int bytes = transport.Read(buffer, offset, remainingCount);
                 if (bytes == 0)
                 {
-                    if (tempCount != count)
+                    if (remainingCount != count)
                     {
                         throw new IOException(SR.net_io_eof);
                     }
 
                     return 0;
                 }
 
-                tempCount -= bytes;
+                remainingCount -= bytes;
                 offset += bytes;
-            } while (tempCount != 0);
+            } while (remainingCount > 0);
 
+            Debug.Assert(remainingCount == 0);
             return count;
         }
 
-        //
-        // Completes "_Request" with 0 if 0 bytes was requested or legitimate EOF received.
-        // Otherwise, reads as directed or completes "_Request" with an Exception or throws.
-        //
-        public void AsyncReadPacket(AsyncProtocolRequest request)
+        /// <summary>
+        /// Completes "request" with 0 if 0 bytes was requested or legitimate EOF received.
+        /// Otherwise, reads as directed or completes "request" with an Exception.
+        /// </summary>
+        public static async void ReadPacketAsync(Stream transport, AsyncProtocolRequest request)
```

Comment:
Your call. I was just thinking it might save someone a few minutes of questioning its use here at some point down the road if/when auditing "async void" uses in the codebase.